### PR TITLE
Use npm for test dependancies.

### PR DESCRIPTION
**Pull request by jcvernaleo**
Add package.json with all test dependancies.

Update instructions for runnings tests.

Add .gitignore to ignore node_modules and emacs ~ files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/SmartPool/contracts/issues"
+  },
+  "homepage": "https://github.com/SmartPool/contracts#readme",
+  "dependencies": {
+    "bignumber.js": "^5.0.0",
+    "ethereumjs-testrpc": "^6.0.3",
+    "truffle": "^4.0.1",
+    "web3": "^1.0.0-beta.26"
```
Comment (yaronvel) [inline package.json]: do we really need web3? isn't it part of truffle? moreover, truffle is using earlier version. 0.x.x.

Comment (jcvernaleo) [reply package.json]: truffle does have web3 as one of its dependencies but that is not available to the test scripts when installed though npm (so the tests fail to run with `Error: Cannot find module 'web3'` if web3 isn't a direct dependence of the project.

Comment (jcvernaleo) [reply package.json]: As for truffle, I used the version npm installed and let it pick.  If there is a version you would prefer I can change it but I didn't see any reason to prefer one over the other.

**Comment by jcvernaleo** [comment]
Since I was doing npm stuff already I added one more commit to use a non-standard port so I don't have to stop geth to run the tests.

**Comment by jcvernaleo** [comment]
All of the tests pass now so this one should be good to go.

**Review by yaronvel — APPROVED** [review]
seems like `sudo` is needed for  `npm install`, at least on my ubuntu.

**Comment by yaronvel** [comment]
npm package for installation and testing.

**Comment by jcvernaleo** [comment]
That is very odd about sudo.  I'm also running on Ubuntu and it installs as a normal user just fine (install everything to contracts/node_modules).  Do you have something in your .npmrc or something similar to make it install globally maybe?

**Pull request merged**